contributing/USER_PERSONAS.md

```diff
@@ -0,0 +1,210 @@
+# OpenAssetIO User Personas
+
+> Note: This page is a WIP, please add/suggest omissions through PR
+
+As a project, the majority of OpenAssetIO users are software developers.
+We count a "user" as someone who interacts with the code base and/or
+distribution, rather than with any higher-order functionality exposed by
+an application integrating the API.
+
+-   [Commercial DCC tool author](#commercial-dcc-tool-author)
```

Comment:
🤔 I'd love to just use that, but do you think it's obvious enough? I quite liked having one in-line in the docs to form some last-ditch attempt to avoid Wall-of-Text-Syndrome™